t/t1300-config.sh

```diff
@@ -1766,6 +1766,87 @@ test_expect_success !MINGW '--show-origin blob ref' '
 	test_cmp expect output
 '
 
+
+test_expect_success '--show-scope with --list' '
+	cat >expect <<-EOF &&
+		(global)	user.global=true
+		(global)	user.override=global
+		(global)	include.path=$INCLUDE_DIR/absolute.include
+		(global)	user.absolute=include
+		(local)	user.local=true
+		(local)	user.override=local
+		(local)	include.path=../include/relative.include
+		(local)	user.relative=include
+		(command line)	user.cmdline=true
+	EOF
+	git -c user.cmdline=true config --list --show-scope >output &&
+	test_cmp expect output
+'
+
+test_expect_success '--show-scope with --list --null' '
+	cat >expect <<-EOF &&
+		(global)Quser.global
+		trueQ(global)Quser.override
+		globalQ(global)Qinclude.path
+		$INCLUDE_DIR/absolute.includeQ(global)Quser.absolute
+		includeQ(local)Quser.local
+		trueQ(local)Quser.override
+		localQ(local)Qinclude.path
+		../include/relative.includeQ(local)Quser.relative
+		includeQ(command line)Quser.cmdline
+		trueQ
+	EOF
+	git -c user.cmdline=true config --null --list --show-scope >output.raw &&
+	nul_to_q <output.raw >output &&
+	# The here-doc above adds a newline that the --null output would not
+	# include. Add it here to make the two comparable.
+	echo >>output &&
+	test_cmp expect output
+'
+
+test_expect_success '--show-scope with single file' '
+	cat >expect <<-\EOF &&
+		(local)	user.local=true
+		(local)	user.override=local
+		(local)	include.path=../include/relative.include
+	EOF
+	git config --local --list --show-scope >output &&
+	test_cmp expect output
+'
+
+test_expect_success '--show-scope with --get-regexp' '
+	cat >expect <<-EOF &&
+		(global)	user.global true
+		(local)	user.local true
+	EOF
+	git config --show-scope --get-regexp "user\.[g|l].*" >output &&
+	test_cmp expect output
+'
+
+test_expect_success '--show-scope getting a single key' '
+	cat >expect <<-\EOF &&
+		(local)	local
+	EOF
+	git config --show-scope user.override >output &&
+	test_cmp expect output
+'
+
+test_expect_success '--show-scope with --show-origin' '
+	cat >expect <<-EOF &&
+		(global)	file:$HOME/.gitconfig	user.global=true
+		(global)	file:$HOME/.gitconfig	user.override=global
+		(global)	file:$HOME/.gitconfig	include.path=$INCLUDE_DIR/absolute.include
+		(global)	file:$INCLUDE_DIR/absolute.include	user.absolute=include
+		(local)	file:.git/config	user.local=true
+		(local)	file:.git/config	user.override=local
+		(local)	file:.git/config	include.path=../include/relative.include
+		(local)	file:.git/../include/relative.include	user.relative=include
```

Comment:
So this is what you referenced [here](https://github.com/git-for-windows/git/pull/2399#issuecomment-559504469) regarding the backtrace?

I am afraid that I was not quite clear what I meant: when I include a file in the `--system` config, and then include another one in a conditional include, and that has the setting that I am looking for, I would like to have that entire trace: the line in the system config that included the intermediate config, and the line in that file that conditionally included the "inner" config, and then the line in that config that set this.

I do agree, though, that this is safely outside the scope of what you have in mind.

Reply:
Oh yeah, in that case definitely out of scope